src/sagemaker_xgboost_container/algorithm_mode/serve.py

```diff
@@ -93,6 +93,10 @@ def predict(cls, data, content_type='text/x-libsvm', model_format='pkl_format'):
                 if y > x + 1:
                     raise ValueError('Feature size of libsvm inference data {} is larger than '
                                      'feature size of trained model {}.'.format(y, x))
+            elif content_type == 'application/x-recordio-protobuf':
+                if y > x + 1:
```

Comment:
I think this is actually the incorrect check, since the scipy sparse matrix seems to keep the full dimensions of the data.